gpumon/gpumon_interface.ml

```diff
@@ -12,60 +12,147 @@
  * GNU Lesser General Public License for more details.
  *)
 
+open Rpc
+open Idl
+
 let service_name = "gpumon"
 let queue_name = Xcp_service.common_prefix ^ service_name
 let xml_path = "/var/xapi/" ^ service_name
 
+(** Uninterpreted string associated with the operation *)
 type debug_info = string
+[@@deriving rpcty]
+
+(* Domain ID of VM *)
 type domid = int
+[@@deriving rpcty]
+
+(** Reason for incompatibility *)
+type incompatibility_reason =
+  | Host_driver
+  | Guest_driver
+  | GPU
+  | Other
+[@@deriving rpcty]
 
-type incompatibility_reason = Host_driver | Guest_driver | GPU | Other
-type compatibility = Compatible | Incompatible of incompatibility_reason list
+(** Boolean: compatible? *)
```

Comment:
This doctstring is a bit weird. This is the compatibility information for vgpu migrations returned from the Nvidia library. Vgpu and pgpu can be compatible or they can be incompatible for a number of reasons. This type encodes that behaviour

Reply:
Yeah, I wasn't sure what to write there.